jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/settings/component/binder/AbstractGridSettingsBinder.java

```diff
@@ -113,7 +116,8 @@ public boolean saveSettings(V component, S settings) {
             setSortOrderToSettings(sortOrder, settings);
             changed = true;
         }
-        List<? extends Grid.Column<?>> componentColumns = getOrderedColumns(component);
+        List<? extends Grid.Column<?>> componentColumns = getOrderedColumns(component).stream()
+                .filter(c -> !isNull(c.getKey())).toList();
```

Comment:
1. `java.util.Objects#nonNull`
2. This method exists to be used as a `java.util.function.Predicate`, `filter(Objects::nonNull)`. In your case, you have to call `c.getKey()` so using `Objects` makes no sense.
3. As discussed, we need to log **warning** that column is ignored.

Reply:
changed and added debug logging